### PR TITLE
Build envoy in gloo repo

### PR DESCRIPTION
**Pull request by yuval-k**
build envoy using gloo

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -24,6 +24,8 @@
 #   go-tests = true
 #   unused-packages = true
 
+ignored = ["github.com/envoyproxy/data-plane-api*"]
+
```
Comment (talnordan) [inline Gopkg.toml]: Nit: Remove excessive empty line.

File: Gopkg.toml [hunk 2]
```diff
@@ -77,13 +79,9 @@
...
   version = "0.5.0"
 
-[[constraint]]
-  version = "v1.1.0"
-  name = "github.com/golang/protobuf"
-
 [[constraint]]
   branch = "master"
-  name = "github.com/grpc-ecosystem/go-grpc-middleware"
+name = "github.com/golang/protobuf"
```
Comment (talnordan) [inline Gopkg.toml]: Nit: indentation

File: Gopkg.toml [hunk 3]
```diff
@@ -201,6 +199,7 @@
   name = "k8s.io/client-go"
   version = "6.0.0"
 
+
```
Comment (talnordan) [inline Gopkg.toml]: Nit: excessive newline

File: build-envoy/BUILD [hunk 4]
```diff
@@ -0,0 +1,23 @@
+licenses(["notice"])  # Apache 2
+
+load(
+    "@envoy//bazel:envoy_build_system.bzl",
+    "envoy_cc_binary",
+    "envoy_cc_library",
+    "envoy_cc_test",
```
Comment (talnordan) [inline build-envoy/BUILD]: Nit: Are "library" and "test" necessary?

File: build-envoy/BUILD [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "envoy_package",
+)
+
+envoy_package()
+
+
+# allow a non stamped version till we figure out why stamping doesn't work..
```
Comment (talnordan) [inline build-envoy/BUILD]: Nit: Mark this comment as an assigned TODO.

File: build-envoy/README.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+To add a new filter:
+1. add it to the WORKSPACE
+2. add it to tests.sh
+3. add it to build-config/extensions_buiold_config.bzl
```
Comment (talnordan) [inline build-envoy/README.md]: Nit: s/add/Add

File: build-envoy/WORKSPACE [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    name = "solo_envoy_common",
+    path = "./envoy-common",
+)
+
+local_repository(
+    name = "transformation_filter",
+    path = "../pkg/plugins/common/transformation/envoy",
+)
+
+
```
Comment (talnordan) [inline build-envoy/WORKSPACE]: Nit: Excessive newline.

**Comment by yuval-k** [comment]
@talnordan @ilackarms this is rebased off master, PTAL.

This builds both on CI and on my local fedora  (with protoc 3.5)

**Pull request merged**